worker/handler/handler.go

```diff
@@ -277,6 +360,20 @@ func (h *Handler) RunStart() (ch *sb.Channel, err error) {
 
 	// create sandbox if needed
 	if h.sandbox == nil {
+		// if estimated mem usage exceeds the hard limit, the handler would queue itself
+        hms.sbCreatingLock.Lock()
+        for hms.sbHardLimit != 0 && (getMemUsage() + (hms.numSbCreating + 1) * hms.avgHandlerMemUsage > hms.sbHardLimit) {
```

Comment:
I don't understand the purpose of having both this and the scheduler that decides when to release blocked RunStarts.

Reply:
Removed in the new commit.